File: luajit_programs_and_containers/building_blocks/scan_tree/scan_tree_c/dsl_tests/pump_test/user_functions.c
#include "scan_tree.h"

uint8_t user_vft_motor_health_check(uint8_t *state, const st_handle_t *h, const st_input_desc_t *inputs, uint32_t n_inputs)
{
    (void)state; (void)n_inputs;
    float cur = ((const float*)st_buf_data(h, inputs[0].buf_id))[inputs[0].start];
    float thr = ((const float*)st_buf_data(h, inputs[1].buf_id))[inputs[1].start];
    return (cur < thr) ? 1 : 0;
}

File: luajit_programs_and_containers/building_blocks/scan_tree_luajit/dsl_tests/pump_test/user_functions.lua
--[[
    User VFT implementations for pump station test
--]]

local M = {}

function M.user_vft_motor_health_check(state, nid, h, inputs, n_inputs)
    local cur = h:buf_data(inputs[1].buf_id)[inputs[1].start]
    local thr = h:buf_data(inputs[2].buf_id)[inputs[2].start]
    return (cur < thr) and 1 or 0
end

return M
